clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17132,6 +17132,10 @@ static bool ConvertAPValueToString(const APValue &V, QualType T,
         case BuiltinType::WChar_U: {
           unsigned TyWidth = Context.getIntWidth(T);
           assert(8 <= TyWidth && TyWidth <= 32 && "Unexpected integer width");
+          if (V.getInt() > std::numeric_limits<uint64_t>::max() ||
```

Comment:
Is the right place to fix the original issue? The code right below this seems to be printing in hex, but the bug report shows a long decimal value. Also I'm not sure why _BitInt would got through any of the BuiltinTypes listed in this switch.

Is the later call to V.getInt().toString(Str); on line 17153 the right place?

Reply:
Also if `V.getInt()` was ever larger than `std::numeric_limits<uint64_t>::max())`, the getZExtValue() call below would trigger an assertion.

Reply:
Hi, do you have any idea where is actually the right place to fix the code? I am still looking for it, but if you have found it, it would be really appreciated!

Reply:
> do you have any idea where is actually the right place to fix the code?

IIUC, it should be around

```c++
      V.getInt().toString(Str);
```

as I mentioned earlier in https://github.com/llvm/llvm-project/pull/75902#discussion_r1432050810

Reply:
@ShamrockLee 

thanks for letting me know! I have moved the code overthere, however it looks like there are some failed tests, is it because of my code which causes the error?

Reply:
> thanks for letting me know! I have moved the code overthere, however it looks like there are some failed tests, is it because of my code which causes the error?

It is because the conditional clause should be *above* the integer printing in question.

Reply:
Besides, as @tbaederr and @shafik suggest, printing the hexadecimal representation of the integer is better than avoid printing it. This would be something like.

```c
      if (V.getInt() > std::numeric_limits<uint64_t>::max() ||
          V.getInt() < std::numeric_limits<int64_t>::min()) {
        // Code to print the hexadecimal expression
      } else {
        V.getInt().toString(Str);
      }
```